src/util/createHash.ts

```diff
@@ -1,9 +1,30 @@
-import { createHash } from 'crypto';
+/**
+ * Browser-compatible hash utilities using SubtleCrypto.
+ * Uses the Web Crypto API which is available in all modern browsers.
+ *
+ * Note: SubtleCrypto is async, so sha256 returns a Promise.
+ * Vite aliases this module in place of ./createHash for browser builds.
+ */
 
-export function sha256(str: string) {
-  return createHash('sha256').update(str).digest('hex');
+/**
+ * Compute SHA-256 hash of a string and return hex digest.
+ * Uses the Web Crypto API (SubtleCrypto).
+ *
+ * @param str - The string to hash
+ * @returns Promise resolving to the hex-encoded SHA-256 hash
+ */
+export async function sha256(str: string): Promise<string> {
```

Comment:
I want to make sure that the hashing algorithm is stable and the hashes before are the same as after. Would be nice to create a test case specifically for this.